lib/vehicle.rb

```diff
@@ -6,20 +6,37 @@ class Vehicle
               :make,
               :model,
               :engine
+  attr_accessor :registration_date, :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = nil
+    @plate_type = nil
   end
 
   def antique?
     Date.today.year - @year > 25
   end
 
+  def regular?
+    @engine == :ice
+  end
+
   def electric_vehicle?
     @engine == :ev
   end
+
+  def plate_type!
+    if antique?
+      @plate_type = :antique
+    elsif regular?
+      @plate_type = :regular
+    else
+      @plate_type = :ev
+    end
+  end
```

Comment:
Great job with this method. I like your use of helper methods in the if conditionals.